wp-includes/sqlite-ast/class-wp-sqlite-information-schema-builder.php

```diff
@@ -0,0 +1,1000 @@
+<?php
+
+// @TODO: Remove the namespace and use statements when replacing the old driver.
+namespace WIP;
+
+use WP_MySQL_Lexer;
+use WP_MySQL_Token;
+use WP_Parser_Node;
+
+class WP_SQLite_Information_Schema_Builder {
+	/**
+	 * Tables that emulate MySQL "information_schema".
+	 *
+	 *  - TABLES
+	 *  - VIEWS
+	 *  - COLUMNS
+	 *  - STATISTICS (indexes)
+	 *  - TABLE_CONSTRAINTS (PK, UNIQUE, FK)
+	 *  - CHECK_CONSTRAINTS
+	 *  - KEY_COLUMN_USAGE (foreign keys)
+	 *  - REFERENTIAL_CONSTRAINTS (foreign keys)
+	 *  - TRIGGERS
+	 */
+	const CREATE_INFORMATION_SCHEMA_QUERIES = array(
+		// TABLES
+		"CREATE TABLE IF NOT EXISTS _mysql_information_schema_tables (
+			TABLE_CATALOG TEXT NOT NULL DEFAULT 'def',   -- always 'def'
+			TABLE_SCHEMA TEXT NOT NULL,                  -- database name
+			TABLE_NAME TEXT NOT NULL,                    -- table name
+			TABLE_TYPE TEXT NOT NULL,                    -- 'BASE TABLE' or 'VIEW'
+			ENGINE TEXT NOT NULL,                        -- storage engine
+			VERSION INTEGER NOT NULL DEFAULT 10,         -- unused, in MySQL 8 hardcoded to 10
+			ROW_FORMAT TEXT NOT NULL,                    -- row storage format @TODO - implement
+			TABLE_ROWS INTEGER NOT NULL DEFAULT 0,       -- not implemented
```

Comment:
I can't think of anything better than computing this and other values here upon query – perhaps a view would work for that.

Reply:
@adamziel Yeah, a view sounds like a good idea. In fact, from MySQL 8, all the information schema tables are views on fewer virtual and invisible system tables under the `mysql` database (`mysql.tables`, etc.), which hints that maybe we could store this info in fewer more universal tables, and then just create views, which could also read from other tables like `sqlite_master` and other ones.

That said, in the first phase, I want to cover as much as possible in the most straightforward way because otherwise it could get a bit too complex for me to tackle at once. Once it works and the test coverage is decent, we could further simplify, optimize, etc.